compiler/noirc_evaluator/src/ssa/opt/loop_invariant.rs

```diff
@@ -210,46 +224,38 @@ struct LoopContext {
     ///
     /// Reset for each new loop as the set should not be shared across different outer loops.
     nested_loop_control_dependent_blocks: HashSet<BasicBlockId>,
-    // Indicates whether the current loop has break or early returns
+    /// Indicates whether the current loop has break or early returns
     no_break: bool,
-
-    // Stores whether the current block being processed is control dependent
-    current_block_control_dependent: bool,
-
-    // Tracks whether the current block has a side-effectual instruction.
-    // This is maintained per instruction for hoisting control dependent instructions
-    current_block_impure: bool,
-
-    // Stores whether the current block has known fix upper and lower bounds that
-    // indicate that it is guaranteed to execute at least once.
-    current_block_executes: bool,
 }
 
-impl LoopContext {
-    fn pre_header(&self) -> BasicBlockId {
-        self.pre_header.expect("ICE: Pre-header block should have been set")
-    }
-
-    fn is_loop_invariant(&self, value_id: &ValueId) -> bool {
-        !self.defined_in_loop.contains(value_id) || self.loop_invariants.contains(value_id)
-    }
+struct BlockContext {
+    /// Indicate that this block is the header of the current loop.
+    is_header: bool,
+    /// Stores whether the current block being processed is control dependent.
+    is_control_dependent: bool,
+    /// Tracks whether the current block has a side-effectual instruction.
+    /// This is maintained per instruction for hoisting control dependent instructions
+    is_impure: bool,
+    /// Stores whether the current loop has known fix upper and lower bounds that
```

Comment:
```suggestion
    /// Stores whether the current loop has known fixed upper and lower bounds that
```
nit: leftover from old comment